### DISCUSSION TITLE
Cannot Link Site Field To Content Type

### DISCUSSION THREAD

**Discussion by indieza** [pnp/pnpcore#692] [Q&A]
Hello,

I have a simple issue over linking a site field to SharePoint Content type.
Here i my code which hold the functionality to create a Content Type and assign a fields to it.
```
var contentType = await context.Web.ContentTypes
    .AddAsync(model.Id, model.Name, $"{model.Description}", model.Group);

contentType.Sealed = model.Sealed;
contentType.ReadOnly = model.ReadOnly;
contentType.Hidden = model.Hidden;

foreach (var field in model.Fields)
{
    try
    {
        await field.AddToContentType(context, contentType);
    }
    catch (Exception ex)
    {
        var pnpError = ex.ToString();
    }
}

await contentType.UpdateAsync();
```

In the foreach loop I iterate over a Fields Input Models collection. There each Field Input Model Class implemed own functionality for lining a field to Content Type based on the SharePoint Field type. So I hit same issue for all of the SharePoint fields types.
Here is the functionality for linking a field
```
public override async Task AddToContentType(PnPContext context, IContentType contentType)
{
    var field = await context.Web.Fields.AddTextAsync(this.InternalName, new FieldTextOptions()
    {
        AddToDefaultView = this.AddToDefaultView,
        Description = $"{this.Description}\nPowered by Indieza Point",
        EnforceUniqueValues = this.EnforceUniqueValues,
        Group = this.Group,
        Hidden = this.Hidden,
        Indexed = this.Indexed,
        Options = this.Options,
        Required = this.Required,
        MaxLength = this.MaxLength,
    });

    field.Title = this.Title;

    if (this.DefaultValue != null)
    {
        field.DefaultValue = this.DefaultValue.ToString();
    }

    await field.UpdateAsync();
    await contentType.FieldLinks.AddAsync(field.InternalName, field.Title);
}
```

The message which I received when link a field is with the following content.
`Column 'XXXXXXX' does not exist. It may have been deleted by another user.`
When I go to the SharePoint site the field is there and somehow it cannot be linked.
Any ideas over that?

Regards,
Simeon Valev

**Comment by jansenbe** [comment]
@indieza : think you're seeing #266... SPO REST is not well suited for configuring content types and we'll need to switch our internal implementation to call the CSOM endpoint here. Will try to prioritize this in the coming days.

**Reply by indieza** [reply]
Okay, so as a conclusion there is some issue over the link field logic in the library, right?

Regards,
Simeon Valev

**Reply by jansenbe** [reply]
I did not yet test your code, but I'm quite sure that's still a missing part, something we need to fix

**Reply by indieza** [reply]
@jansenbe: okay I understand, please give me a sign when it is fixed. Thanks

Regards,
Simeon Valev

**Reply by jansenbe** [reply]
@indieza : I've implemented support for adding fields to content types, for a site/web content type you can add an site/web IField reference,   for a content type already added to a list you can add a list IField reference. See the updated docs here: https://pnp.github.io/pnpcore/using-the-sdk/fields-intro.html

Note: as I committed this today the changes will only be available as of tomorrow with the next nightly build.

**Reply by indieza** [reply]
@jansenbe: Perfect, in that case I will come back tomorrow .

Regards,
Simeon Valev
